### DISCUSSION TITLE
using `virtualFocus` to avoid mobile web bug

### DISCUSSION THREAD

**Discussion by eleith** [ariakit/ariakit#2055] [Questions] [answered]
i'm experimenting with using `ComboboxItem` as links and came across an odd bug that i could work around with setting `virtualFocus=false` prop on `Combobox`

unlike the example [Combobox with links](https://ariakit.org/examples/combobox-links), i was using links without `target=_blank`

i noticed that frequently (but not always?), when touching an item, the link would not be followed

after debugging (directly on firefox mobile) i noticed that when the bug was occuring, a [preventDefault](https://github.com/ariakit/ariakit/blob/1937330ffc4db2b11c2007f3983a77ae97fa2a90/packages/ariakit/src/composite/composite-item.tsx#L269) was occasionally being fired by an onBlur handler on `ComboboxItem`

i was not able to determine why this would happen only sometimes as opposed to everytime.

however, given the [conditional](https://github.com/ariakit/ariakit/blob/1937330ffc4db2b11c2007f3983a77ae97fa2a90/packages/ariakit/src/composite/composite-item.tsx#L264), i was able to set `virtualFocus=false` and gaurantee the link will always work when touched on mobile devices.

i don't know if this is a bug or if i need to better understand when and how to use `virtualFocus`. i did notice on the [Combobox multi-select](https://ariakit.org/examples/combobox-multiple) that this prop is used directly...

**Comment by diegohaz** [comment, marked as answer]
Hey! Could you please open an issue with reproduction steps and the browser/OS versions (so we can check on BrowserStack)?

**Reply by eleith** [reply]
i was able to reproduce in a few browsers (firefox/linux, chrome/android), though it was easiest to reproduce on firefox/android. 

i have filed an [issue](https://github.com/ariakit/ariakit/issues/2056)
